posthog/hogql/functions/mapping.py

```diff
@@ -965,6 +965,283 @@ def compare_types(arg_types: list[ConstantType], sig_arg_types: tuple[ConstantTy
     "leadInFrame": HogQLFunctionMeta("leadInFrame", 1, 1),
     # table functions
     "generateSeries": HogQLFunctionMeta("generate_series", 3, 3),
+    # PostgreSQL-style date/time functions
+    "date_part": HogQLFunctionMeta(
+        "if({} = 'year', toYear({}), if({} = 'month', toMonth({}), if({} = 'day', toDayOfMonth({}), if({} = 'hour', toHour({}), if({} = 'minute', toMinute({}), if({} = 'second', toSecond({}), if({} = 'dow', toDayOfWeek({}), if({} = 'doy', toDayOfYear({}), if({} = 'quarter', toQuarter({}), null)))))))))",  # Maps to same implementation as extract
+        2,
+        2,
+        signatures=[
+            ((StringType(), DateTimeType()), IntegerType()),
+            ((StringType(), DateType()), IntegerType()),
+        ],
+    ),
+    "date_trunc": HogQLFunctionMeta(
+        "dateTrunc",
+        2,
+        3,  # Allow optional timezone parameter
+        tz_aware=True,
+        signatures=[
+            ((StringType(), DateTimeType()), DateTimeType()),
+            ((StringType(), DateTimeType(), StringType()), DateTimeType()),
+        ],
```

Comment:
For things like this where we already have the clickhouse variation defined, I'd much rather we had some form of `alias` field on `HogQLFunctionMeta` as opposed to maintaining two versions of the same function (`date_trunc` and `dateTrunc`) - I can _almost_ guarantee these will get out of sync - I mean, they already are:

```
"dateTrunc": HogQLFunctionMeta("dateTrunc", 2, 2),
```

Shiuldn't these have the same signatures and args? It'll be way more confusing having differing signatures of `dateTrunc` vs `date_trunc`